server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -444,22 +451,35 @@ public synchronized void forceNodeShutdown() {
         HttpResponse response = sendToNodeServlet(NodeForceRestartServlet.class, nvps);
         if (response == null) {
             proxyLogger.warning("Node " + getId() + " failed to shutdown and returned a null response.");
-            LOGGER.exiting(false);
+            // stop the polling thread, mark this proxy as down, force this proxy to re-register
+            teardown("it failed to shutdown and must re-register");
```

Comment:
"1" seems the most reasonable to me.

If the node is "healthy" (it did shutdown despite response) it will re-register and cause a new proxy instance to be created. If it's not healthy (didn't really shut down and perhaps is hung or whatever) then we don't want it to be be registered with hub and have it's proxy lying around in a "shutting down" state.